### PR TITLE
Eng 388 reject uninstall when component in use

### PR DESCRIPTION
**Pull request by Kerruba**
Sorry for the huge PR, but for the checkstyle purposes I run the reformat code and that changed 80% of the files.
In this PR I've:
- Introduced an endpoint to get a summary of bundle's installed components that would fail the uninstall
- Tweak the uninstall to work only on bundles which are actually installed and have an `INSTALLED_COMPLETED` associated job
- Changed the EntandoCoreService into EntandoCoreClient and added the methods to retrieve blocking component usage
- Update install tests to use mock of the EntandoCoreClient
- Started refactoring of the component-manager to remove references to Digital-exchange

### CODE REVIEW

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **kerruba**

If they are not already assigned, you can assign the PR to them by writing `/assign @kerruba` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando-k8s%2Fentando-component-manager).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando-k8s/entando-component-manager/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: src/main/java/org/entando/kubernetes/service/digitalexchange/component/EntandoBundleServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        }
+        return bundles.stream().map(this::convertBundleToLegacyComponent).collect(Collectors.toList());
+    }
+
+
+    public EntandoBundle convertBundleToLegacyComponent(EntandoDeBundle bundle) {
+        EntandoBundle dec = EntandoBundle.newFrom(bundle);
+        if (checkIfInstalled(bundle)) {
+            dec.setInstalled(true);
+//            dec.setJob(jobRepository.findFirstByComponentIdAndStatusOrderByStartedAtDesc(
```
Comment (joewhite101) [inline src/main/java/org/entando/kubernetes/service/digitalexchange/component/EntandoBundleServiceImpl.java:122]: remove

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/entando-k8s/entando-component-manager/pull/24#pullrequestreview-397105255" title="Approved">joewhite101</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando-k8s%2Fentando-component-manager).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/entando-k8s/entando-component-manager/blob/master/OWNERS)~~ [joewhite101]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**